core/src/Streamly/Internal/FileSystem/Posix/ReadDir.hsc

```diff
@@ -205,13 +350,13 @@ readDirStreamEither (curdir, (DirStream dirp)) = loop
         -- fromPtrN, but it is not straightforward because the reclen is
         -- padded to 8-byte boundary.
         name <- Array.fromCString (castPtr dname)
-        (isDir, isMeta) <- checkDirStatus curdir dname dtype
-        if isDir
-        then do
-            if isMeta
-            then loop
-            else return (Just (Left (mkPath name)))
-        else return (Just (Right (mkPath name)))
+        gsRes <- checkDirStatus conf curdir dname dtype
+        case gsRes of
+            GSIsRegDir -> return (Just (Left (mkPath name)))
+            GSIsMetaDir ->  loop
+            GSIsNotDir -> return (Just (Right (mkPath name)))
+            -- We can possibly handle the error here
```

Comment:
why this comment? we have already handled/thrown the error.

Reply:
Re-handle it in the context of the function.

Reply:
is there an example case and possible re-handling?